dolfin_adjoint_common/blocks/function.py

```diff
@@ -57,11 +60,28 @@ def evaluate_adj_component(self, inputs, adj_inputs, block_variable, idx,
             )
             adj_output = self.backend.Function(adj_input_func.function_space())
             adj_output.assign(diff_expr)
-            if isinstance(block_variable.output, (AdjFloat, self.backend.Constant)):
+            if isinstance(block_variable.output, AdjFloat):
                 return adj_output.vector().sum()
+            elif isinstance(block_variable.output, self.backend.Constant):
+                R = block_variable.output._ad_function_space(adj_output.function_space().mesh())
+                return self._adj_assign_constant(adj_output, R)
             else:
                 return adj_output.vector()
 
+    def _adj_assign_constant(self, adj_output, constant_fs):
+        # We assume the shape of the constant == shape of the output function.
```

Comment:
Good catch, I was under the assumption that constants in a linear combination would be considered functions since the following does not work in FEniCS:
```
f.assign(Constant(1.0) * Constant(1.0))
```
with the error: `Expected only linear combinations of Functions in the same FunctionSpaces`

But it seems you can safely combine one constant and one function.